From 5178f226baff561a5a50698986eed6bde8c93617 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sun, 13 Apr 2014 13:01:01 -0700
Subject: [PATCH] Simplify choose plugin

Make choose plugin only perform one regex findall instead of possibly two. This also eliminates the need to strip() the result.

diff --git a/plugins/choose.py b/plugins/choose.py
index f478328..7b92e41 100644
--- a/plugins/choose.py
+++ b/plugins/choose.py
@@ -5,14 +5,12 @@ from util import hook
 
 
 @hook.command
-def choose(inp):
-    """choose <choice1>, [choice2], [choice3], [choice4], ... --
-    Randomly picks one of the given choices."""
-
-    c = re.findall(r'([^,]+)', inp)
-    if len(c) == 1:
-        c = re.findall(r'(\S+)', inp)
-        if len(c) == 1:
-            return 'The decision is up to you!'
-
-    return random.choice(c).strip()
+def choose(inp, notice=None):
+    """choose <choice1>, [choice2], [choice3], etc. -- Randomly picks one of the given choices.
+    :type inp: str
+    """
+    choices = re.findall(r'([^,\s]+)', inp)
+    if len(choices) == 1:
+        notice(choose.__doc__)
+        return
+    return random.choice(choices)
-- 
1.9.2

